app/services/deal_analytic_service.rb

```diff
@@ -79,6 +79,220 @@ def set_redemption_count(deal_id, increment = 1)
+          end
+          if !first_redeem
+            num_redeem_array << num_redemption
+          else
+            num_redeem_array << nothing
+          end
+          temp_start_date = temp_start_date + 1.days
+        end
+        deal_array << num_view_array
+        deal_array << num_redeem_array
+        active_deals_array << deal_array
+      end
+      past_deals.each do |ad|
+        temp_start_date = start_date
+        temp_end_date = end_date
+        num_view = 0
+        num_redemption = 0
+        first_view = true
+        first_redeem = true
+        num_view_array = Array.new
+        num_redeem_array = Array.new
+        deal_array = Array.new
+        nothing = [0]
+
+        deal_array << ad.title
+        deal_array << Time.parse(start_date.to_s).to_f * 1000
+        while temp_start_date <= temp_end_date
+
+          num_view = num_view + Viewcount.where(deal_id: ad.id).where(created_at: temp_start_date..temp_start_date.end_of_day).count
+          if num_view != 0 && first_view
+            first_view = false
+          end
+          if !first_view
+            num_view_array << num_view
+          else
+            num_view_array << nothing
+          end
+
+          num_redemption = num_redemption + Redemption.where(deal_id: ad.id).where(created_at: temp_start_date..temp_start_date.end_of_day).count
+          if num_redemption != 0 && first_redeem
+            first_redeem = false
+          end
+          if !first_redeem
+            num_redeem_array << num_redemption
+          else
+            num_redeem_array << nothing
+          end
+          temp_start_date = temp_start_date + 1.days
+        end
+        deal_array << num_view_array
+        deal_array << num_redeem_array
+        past_deals_array << deal_array
+      end
+      expired_deals = past_deals.pluck(:title)
+      array << active_deals_array
+      array << past_deals_array
+      array << expired_deals
+      array
+    end
+
+    # Returns a nested array
+    # array[0] is first deal type. For example array[0] is for all the data of Discounts
+    # array[0][0] gives name of deal type. For example Discounts
+    # array[0][1] gives is the a deal of type discount
+    # array[0][1][0] gives title of deal
+    # array[0][1][1] gives redemption count of that deal
+    # array[0][last number of array] gives total redemption count for that deal type
+    def get_analytics_for_deals_pie_chart(merchant_id)
+      array = Array.new
+      deals = MerchantService.get_all_active_and_past_deals(merchant_id)
+      unique_deal_type = deals.uniq.pluck(:type_of_deal)
+      unique_deal_type.each do |udt|
+        type_array = Array.new
+        type_array << udt
+        total_redemption = 0
+        deal_type = deals.where(:type_of_deal => udt)
+        deal_type.each do |dt|
+          deal_info = Array.new
+          deal_info << dt.title
+          redemption_count = DealAnalytic.where(:deal_id => dt.id).pluck(:redemption_count)[0]
```

Comment:
This line throws error if no data is seeded. Can add a if condition to set redemption count = 0 if unset? @Woonyong92